### PR TITLE
optimize: Send redis command at one time using pipeline

### LINKED ISSUE #3933: ensure that the Redis registry operates reliably
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by lvekee**
### Ⅰ. Describe what this PR did
Send redis command at one time using pipeline
### Ⅱ. Does this pull request fix one issue?
fixes #3933.

### CODE REVIEW

File: discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java [hunk 1]
```diff
@@ -38,10 +38,7 @@
 import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import redis.clients.jedis.Jedis;
-import redis.clients.jedis.JedisPool;
-import redis.clients.jedis.JedisPubSub;
-import redis.clients.jedis.Protocol;
+import redis.clients.jedis.*;
```
Comment (a364176773) [inline discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java:41]: 不允许import *

File: discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java [hunk 2]
```diff
@@ -152,8 +151,10 @@ public void unregister(InetSocketAddress address) {
         NetUtil.validAddress(address);
         String serverAddr = NetUtil.toStringAddress(address);
         try (Jedis jedis = jedisPool.getResource()) {
-            jedis.hdel(getRedisRegistryKey(), serverAddr);
-            jedis.publish(getRedisRegistryKey(), serverAddr + "-" + RedisListener.UN_REGISTER);
+            Pipeline pipelined = jedis.pipelined();
```
Comment (a364176773) [inline discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java:154]: try-with-resources

File: discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java [hunk 3]
```diff
@@ -142,8 +139,10 @@ public void register(InetSocketAddress address) {
         NetUtil.validAddress(address);
         String serverAddr = NetUtil.toStringAddress(address);
         try (Jedis jedis = jedisPool.getResource()) {
-            jedis.hset(getRedisRegistryKey(), serverAddr, ManagementFactory.getRuntimeMXBean().getName());
-            jedis.publish(getRedisRegistryKey(), serverAddr + "-" + RedisListener.REGISTER);
+            Pipeline pipelined = jedis.pipelined();
```
Comment (a364176773) [inline discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java:142]: try-with-resources

File: discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java [hunk 4]
```diff
@@ -162,15 +166,23 @@ public void subscribe(String cluster, RedisListener listener) {
         String redisRegistryKey = REDIS_FILEKEY_PREFIX + cluster;
         LISTENER_SERVICE_MAP.computeIfAbsent(cluster, key -> new ArrayList<>())
                 .add(listener);
-        threadPoolExecutor.submit(() -> {
+        threadPoolExecutor.scheduleAtFixedRate(() -> {
             try {
+                Map<String, String> instances;
                 try (Jedis jedis = jedisPool.getResource()) {
+                    instances = jedis.hgetAll(redisRegistryKey);
+                    if (instances != null && !instances.isEmpty()) {
```
Comment (a364176773) [inline discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java:174]: 这块重复代码抽出来复用吧

File: discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java [hunk 5]
```diff
@@ -141,19 +143,21 @@ static RedisRegistryServiceImpl getInstance() {
     public void register(InetSocketAddress address) {
         NetUtil.validAddress(address);
         String serverAddr = NetUtil.toStringAddress(address);
-        try (Jedis jedis = jedisPool.getResource()) {
-            jedis.hset(getRedisRegistryKey(), serverAddr, ManagementFactory.getRuntimeMXBean().getName());
-            jedis.publish(getRedisRegistryKey(), serverAddr + "-" + RedisListener.REGISTER);
+        try (Pipeline pipelined = jedisPool.getResource().pipelined()) {
```
Comment (a364176773) [inline discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java:146]: jedisPool.getResource()的jedis连接会回收吗?

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**